### PR TITLE
Fix issue where the folder that pkgbuild should output the pkg file to might not exist when it tries to output the pkg file.

### PR DESCRIPTION
**Pull request by AraHaan**
Fixes: https://github.com/dotnet/arcade/issues/8603.

### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by AraHaan** [comment]
After some additional tests it looks like this does not work like I had hoped.

**Comment by AraHaan** [comment]
This does not work like I intended rip.

File: src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.singlerid.targets [hunk 1]
```diff
@@ -409,7 +409,8 @@
     </PropertyGroup>
     
     <MakeDir Directories="$(_OutputPathRoot)" />
-
+    <!-- avoid error where the root folder for the _InstallerFile does not exist: https://github.com/dotnet/arcade/issues/8603. -->
+    <MakeDir Directories="$(PackageOutputPath)" Condition="!Exists($(PackageOutputPath))" />
```
Comment (AraHaan) [inline src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.singlerid.targets:413]:
```suggestion
    <MakeDir Directories="$(ArtifactsNonShippingPackagesDir)" Condition="!Exists($(ArtifactsNonShippingPackagesDir)) AND '$(IsShipping)' == 'false'" />
```

**Comment by AraHaan** [comment]
Now it *should* work.

**Comment by AraHaan** [comment]
image

File: src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.singlerid.targets [hunk 2]
```diff
@@ -409,7 +409,8 @@
     </PropertyGroup>
     
     <MakeDir Directories="$(_OutputPathRoot)" />
-
+    <!-- avoid error where the root folder for the _InstallerFile does not exist: https://github.com/dotnet/arcade/issues/8603. -->
+    <MakeDir Directories="$(ArtifactsNonShippingPackagesDir)" Condition="!Exists($(ArtifactsNonShippingPackagesDir)) AND '$(IsShipping)' == 'false'" />
```
Comment (AraHaan) [inline src/Microsoft.DotNet.Build.Tasks.Installers/build/installer.singlerid.targets:413]:
```suggestion
    <MakeDir Directories="$(ArtifactsNonShippingPackagesDir)" Condition="!Exists($(ArtifactsNonShippingPackagesDir)) AND '$(IsShipping)' == 'false'" />
    <MakeDir Directories="$(ArtifactsShippingPackagesDir)" Condition="!Exists($(ArtifactsShippingPackagesDir)) AND '$(IsShipping)' == 'true'" />

```

**Comment by AraHaan** [comment]
cc: @NikolaMilosavljevic

I think this looks good. With this change it makes it less likely to have errors on MacOS when the non-shipping folder does not exist when making the pre-bundled macos installer packs from sfxproj's just prior to the bundleproj step.

**Comment by AraHaan** [comment]
Sad thing is that I am currently blocked on my own repo until this is pretty much merged in so it would work without issues.

**Review by mmitche — APPROVED** [review]

**Pull request merged**